grizzly/common/fuzzmanager.py

```diff
@@ -0,0 +1,261 @@
+            return
+        raise AttributeError("can't set attribute")
+
+    def cleanup(self):
+        """Cleanup any resources held by this instance.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        if self._sig_filename is not None:
+            rmtree(str(self._sig_filename.parent))
+
+    def iter_crashes(self, quality_filter=None):
+        """Fetch all crash IDs for this FuzzManager bucket.
+        Only crashes with testcases are returned.
+
+        Arguments:
+            quality_filter (int): Filter crashes by quality value (None for all)
+
+        Returns:
+            generator: generator of CrashEntry
+        """
+
+        def _get_results(endpoint, params=None):
+            """
+            Function to get paginated results from FuzzManager
+
+            Args:
+                endpoint (str): FuzzManager REST API to query (eg. "crashes").
+                params (dict): Params to pass through to requests.get
+
+            Returns:
+                generator: objects returned by FuzzManager (as dicts)
+            """
+            LOG.debug("first request to /%s/", endpoint)
+
+            url = "%s://%s:%d/crashmanager/rest/%s/" \
+                % (self._coll.serverProtocol, self._coll.serverHost,
+                   self._coll.serverPort, endpoint)
+
+            response = self._coll.get(url, params=params).json()
+
+            while True:
+                LOG.debug("got %d/%d %s", len(response["results"]), response["count"], endpoint)
+                while response["results"]:
+                    yield response["results"].pop()
+
+                if response["next"] is None:
+                    break
+
+                LOG.debug("next request to /%s/", endpoint)
+                response = self._coll.get(response["next"]).json()
+
+        # Get all crashes for bucket
+        query_args = [
+            ("op", "AND"),
+            ("bucket", self.bucket_id),
+        ]
+        if quality_filter is not None:
+            query_args.append(("testcase__quality", quality_filter))
+        query = json.dumps(dict(query_args))
+
+        n_yielded = 0
+        for crash in _get_results("crashes", params={"query": query, "include_raw": "0"}):
+
+            if not crash["testcase"]:
+                LOG.warning("crash %d has no testcase, skipping", crash["id"])
+                continue
+
+            n_yielded += 1
+            LOG.debug("yielding crash #%d", n_yielded)
+            result = CrashEntry(crash["id"])
+            result._data = crash  # pylint: disable=protected-access
+            yield result
+
+    def signature_path(self):
+        """Download the bucket data from CrashManager.
+
+        Arguments:
+            None
+
+        Returns:
+            Path: Path on disk where signature exists.
+        """
+        if self._sig_filename is not None:
+            return self._sig_filename
+
+        tmpd = Path(mkdtemp(prefix="grizzly-reduce-"))
```

Comment:
Can you do something like `mkdtemp(dir=grz_tmp("reduce"))` instead? This helps keep all the grizzly related temp files in one spot.